agent/directory.go

```diff
@@ -93,18 +96,47 @@ func tryCleanupDirectory(dir string) {
 		return
 	}
 
-	_ = filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+	usr, err := user.Current()
+	if err != nil {
+		grip.Warning(err)
+		return
+	}
+
+	if strings.HasPrefix(dir, usr.HomeDir) {
+		grip.Notice("not cleaning up directory, because it is in the home directory.")
+		return
+	}
+
+	paths := []string{}
+
+	err = filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
```

Comment:
If there's an error sent by `filepath.Walk` to the `WalkFunc`, I think we want to continue. There could be a permissions issue on a subset of files or directories, preventing us from walking them, but we still want to delete everything else.

Reply:
I think that's a worthwhile risk. If we find ourselves in a git repository *but* we don't forward errors, then we'll remove every directory *except* .git

Reply:
sgtm